src/components/Table/GridTable.tsx

```diff
@@ -1000,15 +1000,22 @@ export type GridDataRow<R extends Kinded> = {
   id: string;
   /** A list of parent/grand-parent ids for collapsing parent/child rows. */
   children?: GridDataRow<R>[];
-  /** Whether to pin this sort to the first/last of its parent's children. */
-  pin?: "first" | "last";
+  /** Whether to pin this sort to the first/last of its parent's children. Providing Pin `filter` property will allow pinned rows to be hidden while filtering
```

Comment:
Suggestion: Maybe format this comment b/c it's really long:

```suggestion
  /**
    * Whether to pin this sort to the first/last of its parent's children.
    *
    * By default, pinned rows are always shown/not filtered out, however providing
    * the pin `filter: true` property will allow pinned rows to be hidden
    * while filtering.
    */
```